docs/format_signatures.rst

```diff
@@ -0,0 +1,122 @@
+
+This theme provides two alternative ways to automatically format/indent API
+documentation signatures.
+
+CSS-based function parameter wrapping
+-------------------------------------
+
+There is a CSS-based formatting rule that can be enabled for long function
+signatures that displays each function parameter on a separate line.  This is
+enabled by default, and works fairly well for Python signatures.
+
+.. confval:: html_wrap_signatures_with_css
+
+   Specifies for which `Sphinx domains
+   <https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html>`__
+   this CSS-based formatting is enabled.
+
+   Supported types are:
+
+   :py:obj:`bool`
+       If `True`, enable CSS-based formatting for all domains.  If `False`,
+       disable CSS-based formatting for all domains.
+
+   :py:obj:`list[str]`
+       List of domains for which to enable CSS-based formatting.
+
+   The default value is :python:`True`
+
+.. confval:: html_wrap_signatures_with_css_column_limit
+
+   Maximum signature length before function parameters are displayed on separate
+   lines.
+
+   Only applies to domains for which :confval:`html_wrap_signatures_with_css` is
+   enabled.
+
+   The default value is :python:`70`.
+
+.. rst-example::
+
+   .. py:function:: long_function_signature_example(\
+                      name: int, \
+                      other_param: list[str], \
+                      yet_another: bool = False, \
+                      finally_another: str = "some long string goes here"\
+                    ) -> Tuple[A, Very, Long, Return, Type]
+
+clang-format-based function parameter wrapping
+-----------------------------------------------
+
+There is a more powerful alternative formatting mechanism based on `clang-format
+<https://clang.llvm.org/docs/ClangFormat.html>`__.  This supports C, C++, Java,
+JavaScript, Objective-C, and C#.
+
+This functionality is available as a separate extension included with this
+theme.  To use it, you must include it in your conf.py file and you must also
+set the :confval:`clang_format_signatures_domain_styles` configuration option.
+
+.. code-block:: python
+
+    extensions = [
+        # other extensions...
+        "sphinx_immaterial.format_signatures",
+    ]
+
+    clang_format_signatures_domain_styles = {
+        "cpp": """{
+        BasedOnStyle: LLVM,
+        ColumnLimit: 70,
+        }""",
+    }
+
+.. confval:: clang_format_signatures_domain_styles
+
+   Dictionary that specifies the `clang-format style options
+   <https://clang.llvm.org/docs/ClangFormatStyleOptions.html>`__ for each
+   domain.  Formatting is enabled *only* for domains that are listed.
+
+   .. code-block:: python
+
+      clang_format_signatures_domain_styles = {
+          "cpp": """{
+          BasedOnStyle: Google,
+          BreakInheritanceList: BeforeColon,
+          ColumnLimit: 70,
+          }""",
+      }
+
+   .. tip::
+
+      It is recommended to include ``ColumnLimit: 70`` as a style option.
```

Comment:
Should probably mention that this is the default value for the CSS styling as well. That way there's an explicit distinction between the 2 implemented column limits.

Reply:
Done.